lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -383,7 +383,15 @@ sub update {
                                     machine_id    => $machine->id,
                                     name          => $job_template_name,
                                     test_suite_id => $test_suite->id,
+                                },
+                                {
+                                    key => 'job_templates_product_id_machine_id_name_test_suite_id',
```

Comment:
This relates to `__PACKAGE__->add_unique_constraint([qw(product_id machine_id name test_suite_id)]);` of course. But maybe you can give the constraint an explicit name instead of relying on the default constraint name (which would change if we added/removed a column). It is documented here: https://metacpan.org/pod/DBIx::Class::ResultSource#add_unique_constraint

Reply:
Oh, I had completely not realized that you could name constraints explicitly. That's a very good idea indeed. Thanks!